budgetkey_data_pipelines/pipelines/procurement/tenders/exemptions/add_publisher_urls_resource.py

```diff
@@ -15,11 +17,24 @@
              45: []}
 
 def resource_filter(resource_data, parameters):
+    max_pages = os.environ.get("EXEMPTION_PUBLISHERS_MAX_PAGES", "")
```

Comment:
I believe that in tests you should be able to provide parameters as well, no?
Either way, for doing the initial sync, having to set an environment variable is not that ideal (what if the process fails somewhere in the middle? then we need to redo everything from scratch?).
A more robust way would be to check if the data for this document already exists, and if not then do the scraping. This will allow in the future more correct behaviours, such as periodically re-scraping records that are still "in progress" (i.e data might still be changing) and not scraping records that have been concluded.
Checking if a document exists could be done e.g. by hitting the DB and collecting the list of existing publications per publisher id (iirc it's not such a big list).